test/mocking_classes.py

```diff
@@ -126,6 +126,77 @@ def rand_step(self, tensordict: Optional[TensorDictBase] = None) -> TensorDictBa
         return self.step(tensordict)
 
 
+class MockBatchedLockedEnv(EnvBase):
+    def __init__(self, device, batch_size=None):
+        super(MockBatchedLockedEnv, self).__init__(device=device, batch_size=batch_size)
+        self.action_spec = NdUnboundedContinuousTensorSpec((1,))
+        self.input_spec = CompositeSpec(
+            action=NdUnboundedContinuousTensorSpec((1,)),
+            observation=NdUnboundedContinuousTensorSpec((1,)),
+        )
+        self.observation_spec = CompositeSpec(
+            next_observation=NdUnboundedContinuousTensorSpec((1,))
+        )
+        self.reward_spec = NdUnboundedContinuousTensorSpec((1,))
+        self.counter = 0
+
+    def set_seed(self, seed: int) -> int:
+        assert seed >= 1
+        self.seed = seed
+        self.counter = seed % 17  # make counter a small number
+        self.max_val = max(self.counter + 100, self.counter * 2)
+        return seed_generator(seed)
+
+    def _step(self, tensordict):
+        self.counter += 1
+        n = (
+            torch.full(tensordict.batch_size, self.counter)
```

Comment:
I'm confused: if ti's batch_lokced, the env batch size indicated the size of the output, right?
Why are we taking it from the tensordict? I guess we assume they're the same but from a design perspective it should be env.batch_size no? Perhaps it's because you wand the non-locked class to inherit from this one? If that's the case, can you leave a comment explaining where this comes from?